### ISSUE TITLE
Brisa api method POST can't use json body

### ISSUE THREAD

**Issue by AlbertSabate** [brisa-build/brisa#576]
**Describe the bug**
Api POST can't use body. Brisa breaks.

**To Reproduce**
https://github.com/AlbertSabate/brisa-debug

**Expected behavior**
Should do the flow as expected. request send body -> server get body -> read & parse body -> return data -> client gets data.

**Comment by aralroca** [comment]
I think it is 2 things:

1. The API Routes are thought to be called from the client (web-component). Being already in the server (in a server action), the problem I believe that when doing fetch you have to use the absolute route, not relative.  In this case, you have to use `fetch(new URL('/api', request.url).href` instead of `fetch('/api')`.

3. Yes, there is a bug I guess related to Server Actions that you do a `.json()` without doing `.clone()` somewhere. I am going to investigate about this.

## About the point 1

Instead of using the `request.url` it would be ideal to use the server address and port to ensure that it always travels internally without going out to the internet. I will expose a Server API that returns this information so it will be possible to do this:

```ts
import { getServer } from 'brisa/server';

// ...
const server = getServer();
/* 
DebugHTTPServer {
  address: {
    address: "::",
    family: "IPv6",
    port: 63621,
  },
  development: true,
  fetch: [Function: fetch],
  hostname: "localhost",
  id: "",
  pendingRequests: 3,
  pendingWebSockets: 1,
  port: 63621,
  protocol: "http",
  publish: [Function: publish],
  ref: [Function: ref],
  reload: [Function: reload],
  requestIP: [Function: requestIP],
  stop: [Function: stop],
  subscriberCount: [Function: subscriberCount],
  timeout: [Function: timeout],
  unref: [Function: unref],
  upgrade: [Function: upgrade],
  url: URL {
    href: "http://localhost:63621/",
    origin: "http://localhost:63621",
    protocol: "http:",
    username: "",
    password: "",
    host: "localhost:63621",
    hostname: "localhost",
    port: "63621",
    pathname: "/",
    hash: "",
    search: "",
    searchParams: URLSearchParams {},
    toJSON: [Function: toJSON],
    toString: [Function: toString],
  },
  [Symbol(Symbol.dispose)]: [Function: dispose],
  */
  ```
  
  @AlbertSabate what do you thing about this new API? It would be super simple to support it, and it can provide value to ensure internal travel without going through internet.
 
 ```js 
 fetch(new URL('/api', getServer().url).href
 ```

**Comment by AlbertSabate** [comment]
Sorry, doing the reproduction I missed to add the fix for this. For point 1. How I solved it's like this:

```typescript
const response = await fetch(`${new URL(finalURL).origin}/api`, {});
```

But won't hurt to have that object now that you mention...  Maybe it's nicer to have getServer() instead of this finalURL string.

**Comment by aralroca** [comment]
> Sorry, doing the reproduction I missed to add the fix for this. For point 1. How I solved it's like this:
> 
> ```ts
> const response = await fetch(`${new URL(finalURL).origin}/api`, {});
> ```
> 
> But won't hurt to have that object now that you mention... Maybe it's nicer to have getServer() instead of this finalURL string.

The problem of `finalURL` comes when you have a domain, so this finalURL is the domain that goes outside, to the internet to `fetch`, when you are already on the server.

**Comment by AlbertSabate** [comment]
Okay, I got what you mean now. Yeah it makes a lot of sense not to have to check the dns to get your domain ip, go to internet provider server and then come back to the server from the outside world once you are in the server already, did you know how many apps on the internet does this? 😮‍💨 Very good optimisation.

**Comment by aralroca** [comment]
In the next Brisa version you can use `fetch(new URL('/api', getServer().url).href)` to fetch using the private URL of the server without going outside

**Comment by aralroca** [comment]
@AlbertSabate About the crash when doing .json(), it is to the response, because it does not go directly to the endpoint, it returns 404. 

<img width="719" alt="Screenshot 2024-10-22 at 18 23 22" src="https://github.com/user-attachments/assets/0e41135c-434e-44b7-a788-17c15b5dc871">


I'm going to check why it returns 404. I understand that once you enter the endpoint should work well the request.json() in the same way that works if you make the request from the browser.

**Comment by aralroca** [comment]
@AlbertSabate is related to the "home" of `/api`. Adding `/api/foo.ts` and fetching to `/api/foo` works:

<img width="720" alt="Screenshot 2024-10-22 at 18 27 41" src="https://github.com/user-attachments/assets/1cc7f288-a3b3-4def-8208-9c7bf9d273ba">

Naturally it is not correct. I will fix it.

**Comment by aralroca** [comment]
@AlbertSabate I found the root of the problem:

https://github.com/brisa-build/brisa/blob/e492eb460b20a1208a501013038f0ea6e6e421a5/packages/brisa/src/cli/serve/serve-options.tsx#L251

This is working for all the entrypoints inside `/api` except the "api home". I'm going to fix this. We can't remove the last `/` only to avoid conflicts with pages like `/apifoopage`

**Comment by aralroca** [comment]
@AlbertSabate solved in [0.1.3-canary.2](https://github.com/brisa-build/brisa/releases/tag/0.1.3-canary.2)
